conjure-http/src/server.rs

```diff
@@ -255,17 +269,30 @@ pub trait AsyncWriteBody<W> {
 /// Conjure service endpoints declared with the `server-request-context` tag will be passed a
 /// `RequestContext` in the generated trait.
 pub struct RequestContext<'a> {
-    request_parts: request::Parts,
+    request_parts: MaybeBorrowed<'a, request::Parts>,
     response_extensions: &'a mut Extensions,
 }
 
 impl<'a> RequestContext<'a> {
     // This is public API but not exposed in docs since it should only be called by generated code.
     #[doc(hidden)]
     #[inline]
+    // FIXME remove in favor of borrowed constructor
     pub fn new(request_parts: request::Parts, response_extensions: &'a mut Extensions) -> Self {
```

Comment:
Some jank here but I don't want to make an internal-API breaking change since we upgrade conjure codegen and runtime libraries in separate excavators.